src/main_thread/decrypt/attach_media_keys.ts

```
@@ -92,12 +84,13 @@ export default async function attachMediaKeys(
     return;
   }
   log.info("DRM: Attaching MediaKeys to the media element");
-  emeImplementation
-    .setMediaKeys(mediaElement, mediaKeys)
+  return setMediaKeys(emeImplementation, mediaElement, mediaKeys)
     .then(() => {
       log.info("DRM: MediaKeys attached with success");
     })
     .catch((err) => {
+      // TODO proper error (rely on the `CREATE_MEDIA_KEYS` code or create a new
+      // one?).
```

Will you do that in another PR ?

---

I can also make it here. But I'm unsure if I have to add another error because it could be considered as an API breaking change.

I still could make it part of the `CREATE_MEDIA_KEYS` error as both are close.

---

I now trigger a `CREATE_MEDIA_KEYS_ERROR` in that case.

It's a change of behavior as before the code would have silently failed I guess? 

Yet in many cases it would have mean infinite buffering if the content is encrypted so it could be considered a better result.